plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -286,6 +292,108 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static <T extends Type> TypeCoercer<DecimalType, T> createDecimalToInteger(DecimalType fromType, T toType)
+    {
+        if (fromType.isShort()) {
+            return new ShortDecimalToIntegerCoercer<>(fromType, toType);
+        }
+        return new LongDecimalToIntegerCoercer<T>(fromType, toType);
+    }
+
+    private static class LongDecimalToIntegerCoercer<T extends Type>
+            extends TypeCoercer<DecimalType, T>
+    {
+        private final long minValue;
+        private final long maxValue;
+
+        public LongDecimalToIntegerCoercer(DecimalType fromType, T toType)
+        {
+            super(fromType, toType);
+
+            if (toType.equals(TINYINT)) {
+                minValue = Byte.MIN_VALUE;
+                maxValue = Byte.MAX_VALUE;
+            }
+            else if (toType.equals(SMALLINT)) {
+                minValue = Short.MIN_VALUE;
+                maxValue = Short.MAX_VALUE;
+            }
+            else if (toType.equals(INTEGER)) {
+                minValue = Integer.MIN_VALUE;
+                maxValue = Integer.MAX_VALUE;
+            }
+            else if (toType.equals(BIGINT)) {
+                minValue = Long.MIN_VALUE;
+                maxValue = Long.MAX_VALUE;
+            }
+            else {
+                throw new TrinoException(NOT_SUPPORTED, format("Could not create Coercer from Decimal to %s", toType));
+            }
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            String stringValue = Decimals.toString((Int128) fromType.getObject(block, position), fromType.getScale());
+            try {
```

Comment:
Do not throw exceptions for invalid values that are being coerced.
Simply fallback to `null`

https://github.com/trinodb/trino/blob/befa877c1def26dc17cbcb96e686c184d352a67b/plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/VarcharToIntegerNumberCoercer.java#L62-L76

Reply:
The  idea is to support the same behaviour as hive or throw.
I will add more details to the pr description then